### PR TITLE
[Merged by Bors] - feat(IntermediateValue): add `IsFixedPt` corollaries

### PR DESCRIPTION
**Pull request by urkud**
Motivated by https://github.com/b-mehta/sharkovsky

Also move the definition of `Function.IsFixedPt` to `Logic/Function/Defs`.

---

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

File: Mathlib/Logic/Function/Defs.lean [hunk 1]
```diff
@@ -202,4 +202,7 @@ protected theorem LeftInverse.id {g : β → α} {f : α → β} (h : LeftInvers
 protected theorem RightInverse.id {g : β → α} {f : α → β} (h : RightInverse g f) : f ∘ g = id :=
   funext h
 
+/-- A point `x` is a fixed point of `f : α → α` if `f x = x`. -/
```
Comment (grunweg) [inline Mathlib/Logic/Function/Defs.lean:205]: Thanks for moving this!

File: Mathlib/Topology/Order/IntermediateValue.lean [hunk 2]
```diff
@@ -486,10 +484,32 @@ theorem intermediate_value_Icc' {a b : α} (hab : a ≤ b) {f : α → δ}
   isPreconnected_Icc.intermediate_value (right_mem_Icc.2 hab) (left_mem_Icc.2 hab) hf
 
 /-- **Intermediate Value Theorem** for continuous functions on closed intervals, unordered case. -/
-theorem intermediate_value_uIcc {a b : α} {f : α → δ} (hf : ContinuousOn f (uIcc a b)) :
-    uIcc (f a) (f b) ⊆ f '' uIcc a b := by
+theorem intermediate_value_uIcc {a b : α} {f : α → δ} (hf : ContinuousOn f [[a, b]]) :
+    [[f a, f b]] ⊆ f '' uIcc a b := by
```
Comment (grunweg) [inline Mathlib/Topology/Order/IntermediateValue.lean:488]: Is there a reason you're not writing `f '' ([[a, b]])` on the right hand side? The fact that you need brackets?

**Review by grunweg — COMMENTED** [review]
Nice, thanks!
maintainer merge

**Comment by riccardobrasca** [comment]
Thanks!

bors merge

File: Mathlib/Topology/Order/IntermediateValue.lean (hunk 2 above)
Comment (urkud) [reply Mathlib/Topology/Order/IntermediateValue.lean:488]: The only reason is that I was writing it at about 1am and missed this `uIcc`.

**Pull request closed without merging**